### PR TITLE
allow contentEndpoint that is rendered on demand and can have variabl…

### PR DESCRIPTION
**Pull request by jerryorr**
…e width

### CODE REVIEW

**Comment by jerryorr** [comment]
So the point of this change is this: https://basecamp.com/1824967/projects/4444406/messages/23008298?enlarge=153773978#attachment_153773978

After selecting certain items in the menu, we need to slide in the trees to pick a node. I had to make these changes so that the width could be variable and animated.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bootslide",
-  "version": "0.4.0",
+  "version": "0.5.0",
```
Comment (nathanbowser) [inline package.json]: Leave the version as 0.4.0 and it will automatically increase when I tag it.

File: src/bootslide.js [hunk 2]
```diff
@@ -68,12 +68,26 @@ Bootslide.prototype.resetMargins = function (ctx) {
...
+
+
   this.emit('slide', target.index(), target)
   $('.bootslide-menu-slider').css('transform', 'translate(' + target.index() * this.width * -1 + 'px,0)')
-
-  // Now animate the height
+  
+  // Now animate the height and width
   $('.bootslide-container').height(target.height())
+  $('.bootslide-container').width(target.width())
```
Comment (nathanbowser) [inline src/bootslide.js]: You could clean this up by chaining it.

File: src/bootslide.js [hunk 3]
```diff
@@ -107,7 +131,27 @@ Bootslide.prototype.buildSections = function (menu, sections, back) {
...
+
+      function render (e) {
+        if (newSection.hasClass('bootslide-content-placeholder')) {
+          newSection.removeClass('bootslide-content-placeholder')
+          newSection.append(target.contentEndpoint())
+          newSection.css('width', '')
+        }
+      }
+
+      a.click(render)
```
Comment (nathanbowser) [inline src/bootslide.js]: What do you think about making render anonymous?

File: src/bootslide.js (hunk 2 above)
Comment (jerryorr) [inline src/bootslide.js]: Yup.

File: src/bootslide.js (hunk 3 above)
Comment (jerryorr) [inline src/bootslide.js]: I feel like there was a reason it was a named function at some point, but that clearly is no longer the case.

File: package.json [hunk 4]
```diff
@@ -8,12 +8,14 @@
     "url": "https://github.com/SpiderStrategies/bootslide"
   },
   "scripts": {
-    "bundle": "node_modules/.bin/browserify -r ./src/bootslide:bootslide -r jquery -o example/bundle.js"
+    "bundle": "node_modules/.bin/watchify -r ./src/bootslide:bootslide -r jquery -o example/bundle.js",
```
Comment (nathanbowser) [inline package.json]: I think bundle should just bundle.

Comment (jerryorr) [inline package.json]: Oops, that was a mistake. I must have typed watchify on the wrong line.

**Comment by jerryorr** [comment]
Changes made.

**Pull request merged**